crates/apollo-mcp-server/src/operations.rs

```diff
@@ -433,6 +433,61 @@ pub fn operation_defs(
     Ok(Some((document, operation, comments.map(|c| c.to_string()))))
 }
 
+pub fn extract_and_format_comments(comments: Option<String>) -> Option<String> {
+    comments.and_then(|comments| {
+        let content = Regex::new(r"(\n|^)\s*#")
+            .ok()?
+            .replace_all(comments.as_str(), "$1");
+        let trimmed = content.trim();
+
+        if trimmed.is_empty() {
+            None
+        } else {
+            Some(trimmed.to_string())
+        }
+    })
+}
+
+pub fn variable_description_overrides(
+    source_text: &str,
+    operation_definition: &Node<OperationDefinition>,
+) -> HashMap<String, String> {
+    let mut argument_overrides_map: HashMap<String, String> = HashMap::new();
+    let mut last_offset = operation_definition
+        .name
+        .clone()
+        .and_then(|n| n.location())
+        .and_then(|span| {
+            source_text[span.end_offset()..]
+                .find('(')
```

Comment:
I realized an edge case here, this will break if theres a comment before the variables that includes a (, which is totally valid
```
query MyQuery # This is a query (not a hat)
($arg: ID) {
  __typename
}
```

I think we can fix it with ✨ regular expressions ✨  but I'll leave it up to you if you want to fix that in this PR or the next one! at a minimum can we add a test that shows how it's failing?

Reply:
can we also add a test that doesn't have variables but does have a field with arguments? I'm not sure if we haven't any tests like that yet, I think that should be fine because the result of this is ignore, but just to make sure!

```
query Something {
  field(id: "123") 
}
```